pkgs/development/libraries/libgcrypt/default.nix

```diff
@@ -19,7 +19,9 @@ stdenv.mkDerivation rec {
   # The build enables -O2 by default for everything else.
   hardeningDisable = stdenv.lib.optional stdenv.cc.isClang "fortify";
 
-  buildInputs = [ libgpgerror ]
+  nativeBuildInputs = [ buildPackages.stdenv.cc ];
+  # libgpgerror.dev needed so configure script finds libgpgcrypt-config.
```

Comment:
That is odd. Specifying `buildInputs = [ libgpgerror ]` should be exactly equivalent to specifying the `.dev` output. (There is a `chooseDevOutputs` or something mapped over all `buildInputs` etc. in stdenv).